include/grpcpp/security/tls_certificate_provider.h

```diff
@@ -127,13 +131,17 @@ class GRPCXX_DLL FileWatcherCertificateProvider final
 
   grpc_tls_certificate_provider* c_provider() override { return c_provider_; }
 
-  // Returns an OK status if the following conditions hold:
+  // Returns an OK status if the both of the following conditions hold:
   // - the currently-loaded root certificates, if any, consist of one or more
   //   valid PEM blocks, and
   // - every currently-loaded identity key-cert pair, if any, has a certificate
   //   chain that consists of valid PEM blocks and has a private key is a valid
   //   PEM block.
+  // Separate validation APIs are provided to validate the above conditions
+  // individually.
   absl::Status ValidateCredentials() const;
+  absl::Status ValidateRootCertificates() const;
```

Comment:
Just thinking - I'm wondering if we should've made this a fn on the interface instead? The amount of duplication does bother me, but then we'd be forcing all impls to have this validation implementation.
And, of our two impls we have similar looking validation code, but it's not guaranteed that is the case, so I guess there's not a great way around it? WDYT?

Reply:
This is what we originally did in #37565, but Mark preferred that it be per-implementation.